extensions/amp-animation/0.1/test/test-css-expr.js

```diff
@@ -0,0 +1,283 @@
+
+  it('should parse number', () => {
+    expect(parsePseudo('0')).to.equal('NUM<0>');
+    expect(parsePseudo('1')).to.equal('NUM<1>');
+    expect(parsePseudo('10')).to.equal('NUM<10>');
+    expect(parsePseudo('0.5')).to.equal('NUM<0.5>');
+    expect(parsePseudo('.5')).to.equal('NUM<0.5>');
+    expect(parsePseudo('1.5')).to.equal('NUM<1.5>');
+    expect(parsePseudo('-0.5')).to.equal('NUM<-0.5>');
+    expect(parsePseudo('-.5')).to.equal('NUM<-0.5>');
+    expect(parsePseudo('+.5')).to.equal('NUM<0.5>');
+    expect(parsePseudo('+1.5')).to.equal('NUM<1.5>');
+    expect(parsePseudo('1e2')).to.equal('NUM<100>');
+    expect(parsePseudo('.1e2')).to.equal('NUM<10>');
+    expect(parsePseudo('1e+2')).to.equal('NUM<100>');
+    expect(parsePseudo('1e-2')).to.equal('NUM<0.01>');
+  });
+
+  it('should parse percent', () => {
+    expect(parsePseudo('0%')).to.equal('PRC<0>');
+    expect(parsePseudo('1%')).to.equal('PRC<1>');
+    expect(parsePseudo('0.5%')).to.equal('PRC<0.5>');
+    expect(parsePseudo('.5%')).to.equal('PRC<0.5>');
+    expect(parsePseudo('1e2%')).to.equal('PRC<100>');
+  });
+
+  it('should parse length', () => {
+    expect(parsePseudo('100px')).to.equal('LEN<100 PX>');
+    expect(parsePseudo('-100px')).to.equal('LEN<-100 PX>');
+    expect(parsePseudo('+100px')).to.equal('LEN<100 PX>');
+    expect(parsePseudo('100.5px')).to.equal('LEN<100.5 PX>');
+    expect(parsePseudo('0.5px')).to.equal('LEN<0.5 PX>');
+    expect(parsePseudo('.5px')).to.equal('LEN<0.5 PX>');
+
+    // Non-px units:
+    expect(parsePseudo('100em')).to.equal('LEN<100 EM>');
+    expect(parsePseudo('100rem')).to.equal('LEN<100 REM>');
+    expect(parsePseudo('100vh')).to.equal('LEN<100 VH>');
+    expect(parsePseudo('100vw')).to.equal('LEN<100 VW>');
+    expect(parsePseudo('100vmin')).to.equal('LEN<100 VMIN>');
+    expect(parsePseudo('100vmax')).to.equal('LEN<100 VMAX>');
+    expect(parsePseudo('100cm')).to.equal('LEN<100 CM>');
+    expect(parsePseudo('100mm')).to.equal('LEN<100 MM>');
+    expect(parsePseudo('100q')).to.equal('LEN<100 Q>');
+    expect(parsePseudo('100in')).to.equal('LEN<100 IN>');
+    expect(parsePseudo('100pc')).to.equal('LEN<100 PC>');
+    expect(parsePseudo('100pt')).to.equal('LEN<100 PT>');
+  });
+
+  it('should parse angle', () => {
+    expect(parsePseudo('10deg')).to.equal('ANG<10 DEG>');
+    expect(parsePseudo('-10deg')).to.equal('ANG<-10 DEG>');
+    expect(parsePseudo('+10deg')).to.equal('ANG<10 DEG>');
+    expect(parsePseudo('1.5deg')).to.equal('ANG<1.5 DEG>');
+    expect(parsePseudo('0.5deg')).to.equal('ANG<0.5 DEG>');
+    expect(parsePseudo('.5deg')).to.equal('ANG<0.5 DEG>');
+
+    // Non-deg units:
+    expect(parsePseudo('10rad')).to.equal('ANG<10 RAD>');
+    expect(parsePseudo('10grad')).to.equal('ANG<10 GRAD>');
+  });
+
+  it('should parse time', () => {
+    expect(parsePseudo('10ms')).to.equal('TME<10 MS>');
+    expect(parsePseudo('10s')).to.equal('TME<10 S>');
+  });
+
+  it('should parse url', () => {
```

Comment:
I believe validator has special rules for valid values of `url`, need to apply the same rules, either at parse or eval time (being being passed through)